contracts/script/utils/IncredibleSquaringDeploymentLib.sol

```diff
@@ -0,0 +1,418 @@
+    ) internal returns (DeploymentData memory) {
+        string memory fileName = string.concat(directoryPath, vm.toString(chainId), ".json");
+
+        require(vm.exists(fileName), "Deployment file does not exist");
+
+        string memory json = vm.readFile(fileName);
+
+        DeploymentData memory data;
+        data.incredibleSquaringServiceManager =
+            json.readAddress(".addresses.IncredibleSquaringServiceManager");
+        data.incredibleSquaringTaskManager =
+            json.readAddress(".addresses.IncredibleSquaringTaskManager");
+        data.registryCoordinator = json.readAddress(".addresses.registryCoordinator");
+        data.operatorStateRetriever = json.readAddress(".addresses.operatorStateRetriever");
+        data.stakeRegistry = json.readAddress(".addresses.stakeRegistry");
+        data.strategy = json.readAddress(".addresses.strategy");
+        data.token = json.readAddress(".addresses.token");
+        data.slasher = json.readAddress(".addresses.instantSlasher");
+
+        return data;
+    }
+
+    /// write to default output path
+    function writeDeploymentJson(
+        DeploymentData memory data
+    ) internal {
+        writeDeploymentJson("script/deployments/incredible-squaring/", block.chainid, data);
+    }
+
+    function writeDeploymentJson(
+        string memory outputPath,
+        uint256 chainId,
+        DeploymentData memory data
+    ) internal {
+        address proxyAdmin =
+            address(UpgradeableProxyLib.getProxyAdmin(data.incredibleSquaringServiceManager));
+
+        string memory deploymentData = _generateDeploymentJson(data, proxyAdmin);
+
+        string memory fileName = string.concat(outputPath, vm.toString(chainId), ".json");
+        if (!vm.exists(outputPath)) {
+            vm.createDir(outputPath, true);
+        }
+
+        vm.writeFile(fileName, deploymentData);
+        console2.log("Deployment artifacts written to:", fileName);
+    }
+
+    function _generateDeploymentJson(
+        DeploymentData memory data,
+        address proxyAdmin
+    ) private view returns (string memory) {
+        return string.concat(
+            '{"lastUpdate":{"timestamp":"',
+            vm.toString(block.timestamp),
+            '","block_number":"',
+            vm.toString(block.number),
+            '"},"addresses":',
+            _generateContractsJson(data, proxyAdmin),
+            "}"
+        );
+    }
+
+    function _generateContractsJson(
+        DeploymentData memory data,
+        address proxyAdmin
+    ) private view returns (string memory) {
+        return string.concat(
+            '{"proxyAdmin":"',
+            proxyAdmin.toHexString(),
+            '","IncredibleSquaringServiceManager":"',
+            data.incredibleSquaringServiceManager.toHexString(),
+            '","incredibleSquaringServiceManagerImpl":"',
+            data.incredibleSquaringServiceManager.getImplementation().toHexString(),
+            '","IncredibleSquaringTaskManager":"',
```

Comment:
Same for this one

Reply:
https://github.com/Layr-Labs/incredible-squaring-avs/pull/120/commits/610a6422e9dcbc53c49a58b8112affae41a646e4